### DISCUSSION TITLE
中医药知识问答助手

### DISCUSSION THREAD

**Discussion by xiaomile** [InternLM/Tutorial#427] [project]
# 为什么要做这个
中医药知识是我国传承了几千年的知识瑰宝
中医药种类繁多，构建知识助手可帮助想学习中医药知识的同学快速入门，亦可对一些常见病（如发烧感冒等）开出药方
[原项目](https://github.com/Real-TomHy/ChineseMedicalQA)是2020年做的，受当时的技术限制（大语言模型技术尚未普及），作者使用关键词匹配实体命名数据集的方式来确认问题类型，然后再从事先建好的图数据库中根据中药名称和问题类型查找答案，从而实现问答，但据作者在其博客中所言效果并不理想。

# 核心思想
主要是使用大模型这块来提炼出用户想查询的中药名称和属性（别名，所属科目，气味，功效等等）

# 所需工作
 - ## 知识库来源
   使用原项目的知识数据集即可
   实现方式
    - RAG外挂知识库
    - 图数据库（neo4j）
   
   两者看效果

 - ## 基座模型
   internlm-chat-7b 
   有条件的话尝试internlm2-chat-7b

 - ## 微调
   让大模型从用户的提问中总结出查询的中药名称和属性，微调数据集从原项目中的关键词来做扩充

-  ## 部署
   使用LMdeploy部署，预计会做量化

 - ## 评测
   了解完opencompass再补充

**Comment by LiyanJin** [comment]
这个项目听上去不错，还可以加入吗？

**Reply by xiaomile** [reply]
可以

**Comment by LiyanJin** [comment]
如何联系？微信？



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年01月22日 18:53 |
| 收件人 | ***@***.***> |
| 抄送至 | Hans ***@***.***>***@***.***> |
| 主题 | Re: [InternLM/tutorial] 中医药知识问答助手 (Discussion #427) |

可以

**Comment by zenith779** [comment]
你好，项目还可以加入吗

**Reply by xiaomile** [reply]
可以加入，发一下你的微信号

**Reply by zenith779** [reply]
jiangzk737

**Comment by LiyanJin** [comment]
你们在“大模型实战营20班”的微信群里吗？ 如在可加我微信，jinliyan2004

**Reply by zenith779** [reply]
不在呢，可发一下具体的微信号不

**Reply by xiaomile** [reply]
你的账号是liyanjin2004吧

**Reply by LiyanJin** [reply]
更正一下，应该是jinliyan2004

**Reply by LiyanJin** [reply]
可添加我微信，jinliyan2004

**Comment by LiyanJin** [comment]
我的账号是liyanjin2004

**Comment by LiyanJin** [comment]
是不是要在主办方提供的表格上登记我们的项目？https://aicarrier.feishu.cn/sheets/EYF4s03B0hm3p1t2WMFcqcUHnwc

**Comment by g29times** [comment]
你好 加一下微信号AGINEO

**Reply by xiaomile** [reply]
目前的任务分配只剩评测了，你确定要加入吗？

**Reply by g29times** [reply]
评测也行。还缺人的话可以拉一下
